Mantle/MTLModel+Difference.m

```diff
@@ -10,11 +10,29 @@ @implementation MTLModel (Difference)
 - (NSArray<NSString *> *)differenceFrom:(id)object {
   MTLModel *actual = (MTLModel *)object;
 
-  NSAssert2([object isKindOfClass:self.class], @"Expected object of type %@, got : %@",
-            self.class, [object class]);
+  if (!actual) {
+    return @[
+      [NSString stringWithFormat:@"Expected: object of type %@, got: nil", NSStringFromClass(self.class)]
+    ];
+  }
 
   NSMutableArray<NSString *> *differences = [NSMutableArray array];
+  if (![actual isKindOfClass:self.class]) {
+    [differences addObject:[NSString stringWithFormat:@"Expected: object of type %@, got: %@",
+                            NSStringFromClass(self.class), NSStringFromClass([object class])]];
+    if (![object isKindOfClass:MTLModel.class]) {
+      return differences;
+    }
+  }
+
   for (NSString *key in self.class.propertyKeys) {
+    if (![actual.class.propertyKeys containsObject:key]) {
+      return @[
+        [NSString stringWithFormat:@"Expected: object of type %@, got: %@",
```

Comment:
I think you need to either:
- Do a full diff between the models, even if they don't have the same property keys. Print a message for every key that's found in one but not the other.
- Don't handle models that aren't exactly of the same type.

I prefer the second one, I think that comparing between objects that don't have the same type doesn't have a lot of value, all you need is to know they don't have the same type.

Anyway if you go with the first, instead of going over `self.class.propertyKeys`, you need to take the union of `propertyKeys` from both of them.

Reply:
dropping it. same class only.